broker-daemon/block-order-worker/index.js

```diff
@@ -573,7 +574,15 @@ class BlockOrderWorker extends EventEmitter {
     // reject the entire block order if an underlying order fails
     osm.once('reject', async () => {
       try {
-        await this.failBlockOrder(blockOrder.id, osm.order.error)
+        if (osm.shouldRetry()) {
+          const validation = async () => {
```

Comment:
probably have to rename this function to something more descriptive about what we're trying to do.